File: gilmorales/frontend/src/vista-usuario-comercial-principal.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';
import '@vaadin/vaadin-avatar/src/vaadin-avatar.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';

@customElement('vista-usuario-comercial-principal')
export class VistaUsuarioComercialPrincipal extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-vertical-layout style="width: 100%; height: 100%; position: fixed; padding: 0px; margin: 0px; flex-direction: column; display: flex; justify-content: flex-start; overflow-y: auto;" id="vaadinVerticalLayout">
 <vaadin-horizontal-layout theme="spacing" id="cabecera" style="align-items: flex-start; flex-shrink: 0; z-index:9999; height: 60px; padding: 0px; margin: 0px; width: 100%;"></vaadin-horizontal-layout>
 <vaadin-vertical-layout id="vaadinVerticalLayout1" style="flex-grow: 1; width: 100%; flex-shrink: 0; justify-content: flex-start; background-color: #ffffff; display: flex; overflow-y: auto; align-items: center; padding: 0px;">
  <vaadin-horizontal-layout id="vaadinHorizontalLayout" style="align-self: stretch; align-items: center; justify-content: flex-start; padding-top: var(--lumo-space-s); flex-shrink: 0; padding-right: var(--lumo-space-m);">
   <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout1" style="flex-grow: 0; justify-content: flex-end; width: 8%; margin-left: var(--lumo-space-xl);">
    <vaadin-avatar id="vaadinAvatar" style="width: 100%; height: 100%; margin-left: var(--lumo-space-xl);"></vaadin-avatar>
   </vaadin-horizontal-layout>
   <vaadin-vertical-layout theme="spacing-xs" id="vaadinVerticalLayout2" style="flex-grow: 0; height: 100%; flex-shrink: 0; padding-left: var(--lumo-space-m);">
    <h2 id="nombreDeUsuario">NombreDeUsuario</h2>
    <h3 id="nombreDeEmpresa">NombreDeEmpresa</h3>
    <vaadin-button id="editarPerffilB">
      Editar perfil 
    </vaadin-button>
    <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout3" style="justify-content: center;">
     <label id="numeroL">0</label>
     <label id="meGustasL">Me gustas</label>
    </vaadin-horizontal-layout>
   </vaadin-vertical-layout>
   <vaadin-horizontal-layout id="vaadinHorizontalLayout2" style="flex-grow: 1; justify-content: flex-end; flex-shrink: 0; margin-left: var(--lumo-space-l);">
    <vaadin-button id="eliminarPublicacion" style="flex-grow: 0;">
      Eliminar publicación 
    </vaadin-button>
   </vaadin-horizontal-layout>
  </vaadin-horizontal-layout>
  <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout4" style="justify-content: flex-start; height: 5%; align-content: center; flex-wrap: wrap; margin-left: var(--lumo-space-m); align-self: stretch;">
   <label id="descripcion" style="flex-grow: 0; flex-shrink: 0; padding-left: var(--lumo-space-l);">Descripción del perfil de la empresa</label>
  </vaadin-horizontal-layout>
  <vaadin-horizontal-layout theme="spacing" id="vaadinHorizontalLayout5" style="width: 10%; justify-content: center; height: 8%; margin-left: var(--lumo-space-m); align-self: stretch;">
   <h2 id="h2">Videos</h2>
  </vaadin-horizontal-layout>
  <hr style="width: 98%; align-self: center; flex-shrink: 0; height: 2px;">
  <vaadin-vertical-layout id="listaVideos" style="flex-grow: 1; width: 100%; flex-shrink: 0; justify-content: flex-start; background-color: #ffffff; display: flex; overflow-y: auto; padding-bottom: 2%; align-items: center; align-self: center;"></vaadin-vertical-layout>
 </vaadin-vertical-layout>
</vaadin-vertical-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: gilmorales/src/main/java/vistas/VistaUsuarioComercialPrincipal.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.html.H2;
import com.vaadin.flow.component.html.H3;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Label;

/**
 * A Designer generated component for the vista-usuario-comercial-principal template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("vista-usuario-comercial-principal")
@JsModule("./src/vista-usuario-comercial-principal.ts")
public class VistaUsuarioComercialPrincipal extends LitTemplate {

    @Id("vaadinVerticalLayout")
	private Element vaadinVerticalLayout;
	@Id("cabecera")
	private HorizontalLayout cabecera;
	@Id("vaadinVerticalLayout1")
	private Element vaadinVerticalLayout1;
	@Id("vaadinHorizontalLayout")
	private HorizontalLayout vaadinHorizontalLayout;
	@Id("vaadinHorizontalLayout1")
	private HorizontalLayout vaadinHorizontalLayout1;
	@Id("fotoDePerfil")
	private Image fotoDePerfil;
	@Id("vaadinVerticalLayout2")
	private Element vaadinVerticalLayout2;
	@Id("nombreDeUsuario")
	private H2 nombreDeUsuario;
	@Id("nombreDeEmpresa")
	private H3 nombreDeEmpresa;
	@Id("editarPerffilB")
	private Button editarPerffilB;
	@Id("vaadinHorizontalLayout2")
	private HorizontalLayout vaadinHorizontalLayout2;
	@Id("eliminarPublicacion")
	private Button eliminarPublicacion;
	@Id("vaadinHorizontalLayout3")
	private HorizontalLayout vaadinHorizontalLayout3;
	@Id("numeroL")
	private Label numeroL;
	@Id("meGustasL")
	private Label meGustasL;
	@Id("vaadinHorizontalLayout4")
	private HorizontalLayout vaadinHorizontalLayout4;
	@Id("descripcion")
	private Label descripcion;
	@Id("vaadinHorizontalLayout5")
	private HorizontalLayout vaadinHorizontalLayout5;
	@Id("h2")
	private H2 h2;
	@Id("listaVideos")
	private Element listaVideos;

	/**
     * Creates a new VistaUsuarioComercialPrincipal.
     */
    public VistaUsuarioComercialPrincipal() {
        // You can initialise any data required for the connected UI components here.
    }

}
